src/extension/src/ProcessHandler.py

```diff
@@ -131,6 +131,16 @@ def stage_auto_assess_sh_safely(self, core_process_command):
             self.logger.log_error("Unable to stage auto-assess shim. [Error={0}]".format(str(error)))
 
     def get_python_cmd(self):
+        # auto-discover what is already being used by extension
+        python_path = os.path.realpath(sys.executable)
+        code, out = self.env_layer.run_command_output("which " + python_path, False, False)
```

Comment:
This will not discover python from /usr/share/oem/python/bin/python3 or /usr/libexec/platform-python or /usr/share/oem/python/bin/python paths. causing handler to eventually fail with python not found error, which was the case found in RHEL 8-gen2

Reply:
Why is that? It's inheriting the path already discovered by the shell script that launched handler.

Reply:
Did not see line 135 where it fetches from what is already used. In that case, we only need to verify, with line 135-136, that python paths are correctly discovered.

Reply:
Good to go then? There is a second PR (left open in parallel) to remove the old code once this is proven in production with a release. If this code fails (0.01% chance), the old code picks up.

Reply:
Yes.